crates/primitives-traits/src/lib.rs

```diff
@@ -92,3 +92,16 @@ pub trait MaybeArbitrary {}
 impl<T> MaybeArbitrary for T where T: for<'a> arbitrary::Arbitrary<'a> {}
 #[cfg(not(any(feature = "test-utils", feature = "arbitrary")))]
 impl<T> MaybeArbitrary for T {}
+
+/// Helper trait that requires de-/serialize implementation since `serde` feature is enabled.
+#[cfg(feature = "serde")]
+pub trait MaybeSerialize: serde::Serialize + for<'de> serde::Deserialize<'de> {}
```

Comment:
since this is ser + de perhaps we should rename this to MaybeSerde?

Reply:
this is the same on main, should be addressed in sep pr

Reply:
https://github.com/paradigmxyz/reth/pull/12612